active/0000-standarise-targets.md

```diff
@@ -0,0 +1,152 @@
+- Start Date: 2017-01-03
+- RFC PR: (leave this empty)
+- Ember CLI Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the introduction of a official convention to specify the targets of an
+application.
+
+### Motivation
+
+Javascript and the platforms it runs on are moving targets. NodeJS and browsers release new
+versions every few weeks. Browsers auto update and each update brings new language features
+and APIs.
+
+Developers need an easy way to express intention that abstracts them from the ever-changing
+landscape of versions and feature matrices, so this RFC proposes to introduce a universal
+configuration option to let developers express they intended targets
+
+This configuration should be easily available to addons, but this RFC doesn't impose
+any mandatory behavior on addons. All addons that want to customize their behavior
+depending on the target platforms will have a single source of truth to get that
+information but it's up to them how to use it.
+
+The advantage of having a single source of truth for the targets compared to configure this
+on a per-addon basis like we do today is mostly for better ergonomics and across-the-board
+consistency.
+
+Addons that would benefit from this conventions are `babel-preset-env`, `autoprefixer`,
+`stylelint` and `eslint` (vía `eslint-plugin-compat`), but not only those. Even Ember itself could,
+once converted into an addon, take advantage of that to avoid polyfilling or even taking
+advantage of some DOM API (`node.classList`?) deep in Glimmer's internals.
+
+### Detailed implementation
+
+
+### Browser support
+
+What seems to be the most popular tool and the state of the art on this is the [browserlist](https://github.com/ai/browserslist)
+npm package.
+
+That package is the one behind `babel-preset-env`, `autoprefixer` and others, and uses the data from
+[Can I Use](http://caniuse.com/) for knowing the JS, CSS and other APIs available on every browser.
+
+The syntax is also pretty flexible as it allows all sort of complex queries like `Firefox >= 20`,
+`>2.5% in CA` (browsers with a market share over 2.5% in Canada) and logical combinations.
+
+Using this syntax would allow us to integrate easily with other tools.
+
+Example usage:
+
+```js
+var app = new EmberApp(defaults, {
+  targets: {
```

Comment:
Actually, one interesting approach of this is the one `eslint-plugin-compat` uses. It reads the `browserlist` property in the package.json.

If we are cool with reading the `engines` property of the package.json for the node support it might also be a reasonable place to the list of supporter browsers.